### PR TITLE
[TFL] Remove (log-)softmax before arg-minmax as (log-)softmax is monotonic

### PR DESCRIPTION
**Pull request by WindQAQ**
It's a common pattern for classification task. Most models in `tf.keras.applications` set `classifier_activation="softmax"` by default. This PR optimizes out that softmax op when users want to get prediction ids only instead of probabilities.

### CODE REVIEW

File: tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td [hunk 1]
```diff
@@ -684,3 +684,20 @@ def OptimizeMul1ToIdentity : Pat<
   (replaceWithValue $input),
   [(HaveSameType $input, $constant),
    (AllElementsAre<"1.0f"> $constant)]>;
+
+// Remove (log-)softmax before arg-minmax as (log-)softmax is monotonic.
```
Comment (abattery) [inline tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td:688]: Can we move these op lowering under lower_tf.cc or lower_tf.td in the tensorflow mlir directory?

Comment (abattery) [reply tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td:688]: There is an optimization pattern for TensorFlow MLIR dialect:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/tensorflow/transforms/optimize.td

Comment (WindQAQ) [reply tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td:688]: Thanks for pointing out! My concern is that there are some patterns that converts TFL ops into TFL_Softmax. See `OptimizeToSoftmax` and `FoldNormalizationIntoSoftmax`. If we put this pattern into TF dialect, it won't take effect for those optimizations cause the ops have been legalized. Is there any workaround?

Comment (abattery) [reply tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td:688]: Do you think `OptimizeToSoftmax` also can be moved to the optimize.td under the TensorFlow MLIR without causing any impacts on the existing models?

Comment (abattery) [reply tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td:688]: If not, I think the current approach will be fine.

Comment (WindQAQ) [reply tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td:688]: I'm afraid it's hard to do that without breaking existing models because there are more ops used in `OptimizeToSoftmax`, such as TFL_Sum and TFL_Exp, and those ops are involved in other patterns. That is, we might have to move more patterns than just one. I'd say that most optimization patterns in TFL are suitable in TF as well (regardless XLA); probably a long-term plan is to migrate those compatible patterns into TF. Thank you!

**Review by abattery — APPROVED** [review]

**Comment by abattery** [comment]
@WindQAQ Could you sync it to head again? There are merge conflicts, which block the submission.

**Comment by WindQAQ** [comment]
Updated!

**Review by abattery — APPROVED** [review]
